src/internal/connector/data_collections.go

```diff
@@ -109,6 +113,57 @@ func verifyBackupInputs(sels selectors.Selector, userPNs, siteIDs []string) erro
 	return nil
 }
 
+// parseMetadataCollections produces two maps:
+// 1- paths: folderID->filePath, used to look up previous folder pathing
+// in case of a name change or relocation.
+// 2- deltas: folderID->deltaToken, used to look up previous delta token
+// retrievals.
+// Finally
+func parseMetadataCollections(
```

Comment:
I'm a little hesitant to have this in this file. It makes the assumptions that
* all data categories will have the same metadata files (or if not requires extra logic)
* all data categories will deserialize their metadata into the same types here and through the entire callstack to the separate category functions

That may hold true or it may not. I'm not sure if I know enough of how things will hash out to say at this point

Reply:
> all data categories will have the same metadata files

This can be addressed if/when differences arise in the future via a pivot on the service/category.  Given how small the scope of usage is at this time, I'd rather start with a smaller footprint that has some hardcoded assumptions than to try and solve the world.

> all data categories will deserialize their metadata into the same types here and through the entire callstack to the separate category functions

As both the producer and consumer of its metadata, GC has the opportunity to make assumptions/assertions such as type constraints without worry of fault.  Standardizing on the broadly expected type used by (de)serialization up front will provide better foundation for future implementation.  .  After that, there's room to expand on how we interface with granular, different assumptions within categories or services.

Reply:
>This can be addressed if/when differences arise in the future via a pivot on the service/category.

Honestly, I'd like to see less of us manually pivoting behavior based on service/category and more of us taking advantage of language features like using interfaces and providing different concrete implementations. The generic code becomes cleaner and it becomes easier to add custom implementations because only the new implementation needs to be written instead of editing the entire call stack to add more pivot options

>As both the producer and consumer of its metadata

I think this is a bit of a red herring. GC is the top-level package for all of this stuff, but it's actually not the producer or consumer of the metadata. The producers and consumers are the individual implementations for each (service, category) pair. Tying the format to the top-level package which only contains broad abstractions seems like it's going to either lead to trying to fit things into the existing (suboptimal) design or cause more refactoring down the road.

I think that perhaps due to the current implementation that manually switches behavior based on the service/category it hasn't been as clear, but really GC (as a struct) is just an abstract entry point to reaching more specific behavior. All the subpackages of the `connector` package could be rewritten to implement a common interface, making GC just something that provides a uniform API and some setup code for other stuff. That would make it clear that GC itself is neither producing or consuming the data, just passing it to other parts of the system

Reply:
> I think this is a bit of a red herring.

Good points.  This func has been moved into `exchange`, and is called here from line ~125.  We can get more idiomatic control here with some additional refactoring in this file.  But for now it should get things to meet up halfway.